src/renderer/grid_renderer.rs

```diff
@@ -217,39 +218,19 @@ impl GridRenderer {
         if self.box_char_renderer.draw_glyph(
             text,
             boxchar_canvas,
-            PixelRect::from_origin_and_size(pos, fragment_size),
+            region,
             style.foreground(&self.default_style.colors).to_color(),
             window_position,
         ) {
             return (text_drawn, true);
         } else if !text.is_empty() {
-            let mut paint = Paint::default();
-            paint.set_anti_alias(false);
-            paint.set_blend_mode(BlendMode::SrcOver);
             text_canvas.save();
 
             // We don't want to clip text in the x position, only the y so we add a buffer of 1
             // character on either side of the region so that we clip vertically but not horizontally.
```

Comment:
I recognize this comment isn't from this PR, but would it not be more correct to clip to the width of the window instead of doing just a single character of buffer zone? I think the logic might be simpler too cause we wouldn't have to reason about adding characters. Idk take it or leave it.